server/src/infra/repositories/partner.repository.ts

```diff
@@ -24,4 +24,11 @@ export class PartnerRepository implements IPartnerRepository {
   async remove(entity: PartnerEntity): Promise<void> {
     await this.repository.remove(entity);
   }
+
+  async update(entity: Partial<PartnerEntity>): Promise<PartnerEntity> {
+    await this.repository.save(entity);
+    return this.repository.findOneOrFail({
+      where: { sharedById: entity.sharedById, sharedWithId: entity.sharedWithId },
+    });
+  }
```

Comment:
This code is basically copy/pasted. Can you extract it to a private method called `.save(entity)`?